internal/weapons/catalyst/hakushin/hakushin.go

```diff
@@ -32,9 +33,12 @@ func NewWeapon(c *core.Core, char *character.CharWrapper, p weapon.WeaponProfile
 	m := make([]float64, attributes.EndStatType)
 	dmg := .075 + float64(r)*.025
 
-	hrfunc := func(ele attributes.Element, key string) func(args ...interface{}) bool {
+	hrfunc := func(ele attributes.Element, key string, gadgetEmit bool) func(args ...interface{}) bool {
```

Comment:
it's weird to have this be the only different implementation of the gadget vs no gadget check

Reply:
This is the only one that uses a cb function generator instead of a cb. I think it has to do with the ICD on Hakushin ring, it can buff while the buff is already active or something,